server/aesdsocket.c

```diff
@@ -358,284 +357,280 @@ static void handle_socket()
+      exit(1);
+    }
+  else if(file_write != strlen(output_buffer))
+    {
+      printf("Error! File is partially written\n");
+      syslog(LOG_ERR,"File is partially written");
+      exit(1);
+    }
+  else
+    {
+      printf("File is written successfully\n");
+      syslog(LOG_INFO,"File is written successfully");
+    }
+
+  printf("Writing data to file %s \n",output_buffer);
+
+  close(file_fd);
+
+  memset(buffer,0,BUFFER_SIZE);
+  file_fd = open(FILE_STORAGE_PATH,O_RDONLY); //open the file in readonly mode
+  if(file_fd == -1)
+    {
+      printf("Error while file reading\n");
+      syslog(LOG_ERR,"Error while file reading");
+      exit(1);
+    }
+  else
+    {
+      printf("File reading succeed\n");
+      syslog(LOG_ERR,"File reading succeed");
+    } 
+
+  //10. Sending data 
+
+  for(i=0;i<received_packet_size;i++)
+    {
+      read_file_data = read(file_fd, &ch, 1); //read the file data
+      if(read_file_data == -1)
         {
-          printf("Error while file reading\n");
-          syslog(LOG_ERR,"Error while file reading");
+          printf("File Read failed\n");
+          syslog(LOG_ERR,"File Read failed");
           exit(1);
         }
       else
         {
-          printf("File reading succeed\n");
-          syslog(LOG_ERR,"File reading succeed");
-        }	
-	
-	//10. Sending data 
-	  
-      for(i=0;i<received_packet_size;i++)
+          //printf("File Read succeed\n");
+          syslog(LOG_INFO,"File Read succeed");
+        }
+      send_data  = send(params->cl_accept_fd,&ch,1,0);  //send
+      if(send_data == -1)
         {
-          read_file_data = read(file_fd, &ch, 1); //read the file data
-          if(read_file_data == -1)
-            {
-              printf("File Read failed\n");
-              syslog(LOG_ERR,"File Read failed");
-              exit(1);
-            }
-          else
-            {
-              //printf("File Read succeed\n");
-              syslog(LOG_INFO,"File Read succeed");
-            }
-          send_data  = send(params->cl_accept_fd,&ch,1,0);  //send
-          if(send_data == -1)
-            {
-              printf("Error in sending file data\n");
-              syslog(LOG_ERR,"Error in sending file data");
-              exit(1);
-            }
-            //printf("Send succeeded\n");
+          printf("Error in sending file data\n");
+          syslog(LOG_ERR,"Error in sending file data");
+          exit(1);
         }
-      close(file_fd);
-	  
-	  mutex_return_val = pthread_mutex_unlock(params->mutex);
-	 if(mutex_return_val != 0)
-	 {
-        printf("pthread_mutex_unlock() has failed\n");
-        syslog(LOG_ERR,"pthread_mutex_unlock() has failed");
-		exit(1);
-	 }
-
-	params->thread_comp_flag = true;
-
-	close(params->cl_accept_fd);
-	  
-    free(output_buffer); // free the buffer
-	
-	return params;
-}
- 
-  //}
-	  
+      //printf("Send succeeded\n");
+    }
+  close(file_fd);
+
+  mutex_return_val = pthread_mutex_unlock(params->mutex);
```

Comment:
@Amey2904dash  aren't you holding mutex for too long? Are you locking code or data?